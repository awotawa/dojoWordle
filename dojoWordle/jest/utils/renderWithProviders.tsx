import { ThemeProvider } from '@emotion/react';
import { ReactElement } from 'react';
import { render } from '@testing-library/react-native';
// import { SafeAreaProvider } from 'react-native-safe-area-context';
import { theme } from '../../src/shared/services/theme/theme';
import React from 'react';

export const renderWithProviders = (component: ReactElement) => {
  render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
};
// <SafeAreaProvider>
// </SafeAreaProvider>
