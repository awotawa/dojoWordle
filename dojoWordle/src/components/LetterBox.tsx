import styled from '@emotion/native';
import React, { useState } from 'react';

type LetterBoxProps = {
  letter: string;
};

export const LetterBox = ({ letter }: LetterBoxProps) => {
  const [validity] = useState<Validity>('nofill');

  return (
    <LetterBoxContainer validity={validity}>
      <TextContainer>{letter}</TextContainer>
    </LetterBoxContainer>
  );
};

type Validity = 'valid' | 'semivalid' | 'invalid' | 'nofill';

const LetterBoxContainer = styled.View<{ validity: Validity }>(
  ({ validity, theme }) => ({
    borderWidth: 2,
    padding: theme.spacing[12],
    borderRadius: theme.spacing[12],
    width: theme.spacing[120],
    height: theme.spacing[120],
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.color[validity],
  })
);

const TextContainer = styled.Text(({ theme }) => ({
  ...theme.typography.title.title1,
}));
