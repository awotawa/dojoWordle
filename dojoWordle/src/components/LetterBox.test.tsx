import { renderWithProviders } from '../../jest/utils/renderWithProviders';
import { screen } from '@testing-library/react-native';
import React from 'react';
import { LetterBox } from './LetterBox';

describe('LetterBox', () => {
  it('should display properly', () => {
    renderWithProviders(<LetterBox letter="B" />);
    expect(screen.getByText('B')).toBeTruthy();
  });
});
