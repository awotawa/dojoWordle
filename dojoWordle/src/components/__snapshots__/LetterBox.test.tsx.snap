// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LetterBox should display properly when it's invalid and be red 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#DC045B",
        "borderRadius": 12,
        "borderWidth": 2,
        "height": 72,
        "justifyContent": "center",
        "padding": 12,
        "width": 72,
      },
      undefined,
    ]
  }
  testID="letterbox"
  validity="invalid"
>
  <Text
    style={
      [
        {
          "fontFamily": "Gotham-Medium",
          "fontSize": 36,
        },
        undefined,
      ]
    }
  >
    M
  </Text>
</View>
`;

exports[`LetterBox should display properly when it's misplaced and be yellow 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#E6CC00",
        "borderRadius": 12,
        "borderWidth": 2,
        "height": 72,
        "justifyContent": "center",
        "padding": 12,
        "width": 72,
      },
      undefined,
    ]
  }
  testID="letterbox"
  validity="misplaced"
>
  <Text
    style={
      [
        {
          "fontFamily": "Gotham-Medium",
          "fontSize": 36,
        },
        undefined,
      ]
    }
  >
    A
  </Text>
</View>
`;

exports[`LetterBox should display properly when it's valid and be green 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#B3CC20",
        "borderRadius": 12,
        "borderWidth": 2,
        "height": 72,
        "justifyContent": "center",
        "padding": 12,
        "width": 72,
      },
      undefined,
    ]
  }
  testID="letterbox"
  validity="valid"
>
  <Text
    style={
      [
        {
          "fontFamily": "Gotham-Medium",
          "fontSize": 36,
        },
        undefined,
      ]
    }
  >
    B
  </Text>
</View>
`;
